protocols/imap/src/test/java/org/apache/james/imap/processor/GetAnnotationProcessorTest.java

```diff
@@ -0,0 +1,233 @@
+        KEYS = ImmutableSet.of("anyKey");
+        annotationRequest = new GetAnnotationRequest(TAG, ImapCommand.anyStateCommand("Name"), ImapConstants.INBOX_NAME);
+        humanTextCaptor = ArgumentCaptor.forClass(HumanReadableText.class);
+        captorResponsecode = ArgumentCaptor.forClass(StatusResponse.ResponseCode.class);
+
+        when(mockImapSession.getState()).thenReturn(ImapSessionState.SELECTED);
+        when(mockImapSession.getAttribute(ImapSessionUtils.MAILBOX_SESSION_ATTRIBUTE_SESSION_KEY)).thenReturn(mailboxSession);
+        when(mockImapSession.getLog()).thenReturn(log);
+    }
+
+    @Before
+    public void setUp() {
+        MockitoAnnotations.initMocks(this);
+        initAndMockData();
+
+        processor = new GetAnnotationProcessor(mockNextProcessor, mockMailboxManager, mockStatusResponseFactory);
+    }
+
+    @Test
+    public void getImplementedCapabilitiesShouldContainSupportAnnotationWhenSupportedByMailboxManager() {
+        assertThat(processor.getImplementedCapabilities(new FakeImapSession())).containsExactly(ImapConstants.SUPPORTS_ANNOTATION);
+    }
+
+    @Test
+    public void processShouldResponseNoWithFailureWhenMailboxDoesNotExist() throws Exception {
+        doThrow(MailboxNotFoundException.class).when(mockMailboxManager).getAllAnnotations(eq(inbox), eq(mailboxSession));
+
+        processor.process(annotationRequest, mockResponder, mockImapSession);
+
+        verify(mockStatusResponseFactory, times(1)).taggedNo(any(String.class), any(ImapCommand.class), humanTextCaptor.capture(), captorResponsecode.capture());
+
+        assertThat(humanTextCaptor.getAllValues()).containsOnly(HumanReadableText.FAILURE_NO_SUCH_MAILBOX);
+        assertThat(captorResponsecode.getAllValues()).containsOnly(StatusResponse.ResponseCode.tryCreate());
+    }
+
+    @Test
+    public void processShouldResponseNoWithGenericFailureWhenManagerThrowMailboxException() throws Exception {
+        doThrow(MailboxException.class).when(mockMailboxManager).getAllAnnotations(eq(inbox), eq(mailboxSession));
+
+        processor.process(annotationRequest, mockResponder, mockImapSession);
+
+        verify(mockStatusResponseFactory, times(1)).taggedNo(any(String.class), any(ImapCommand.class), humanTextCaptor.capture());
+
+        assertThat(humanTextCaptor.getAllValues()).containsOnly(HumanReadableText.GENERIC_FAILURE_DURING_PROCESSING);
+    }
+
+    @Test
+    public void processShouldGetAllAnnotationsAndReturnWithCompleteResponse() throws Exception {
+        when(mockStatusResponseFactory.taggedOk(any(String.class), any(ImapCommand.class), any(HumanReadableText.class)))
+                .thenReturn(okResponse);
+
+        processor.process(annotationRequest, mockResponder, mockImapSession);
+
+        verify(mockMailboxManager, times(1)).getAllAnnotations(inbox, mailboxSession);
+        verify(mockStatusResponseFactory, times(1)).taggedOk(any(String.class), any(ImapCommand.class), humanTextCaptor.capture());
```

Comment:
here it might be interesting to which subclass of ImapCommand it was called with